doc/tutorial/plugins/libdnf5-plugins.rst

```diff
@@ -0,0 +1,128 @@
+        ...
+
+Define the connected CMake option in the ``dnf5/CMakeLists.txt`` file:
+
+.. code-block:: cmake
+
+    option(WITH_PLUGIN_TEMPLATE "Build a dnf5 template plugin" ON)
+
+This sets up the default behavior to include the plugin in the build.
+
+Include the newly created plugin in the ``CMakeLists.txt`` parent file inside
+``libdnf5-plugins``: ``add_subdirectory("template")``.
+
+Delivering to the User
+----------------------
+
+Each plugin has a configuration file where we define the plugin's name
+and specify when to enable it during runtime. Options include:
+
+* ``NO``: Plugin is disabled.
+* ``YES``: Plugin is enabled.
+* ``HOST_ONLY``: Plugin is enabled only in configurations without installroot.
+* ``INSTALLROOT_ONLY``: Plugin is enabled only in configurations with installroot.
+
+Additional configuration options and sections can be defined and accessed from
+the plugin implementation. Here's an example configuration file:
+
+.. literalinclude:: ../../templates/libdnf5-plugin/template.conf
+    :language: cfg
+    :linenos:
+
+In the :ref:`libdnf5 plugins tutorial`, each plugin is delivered in a separate
+package. Include a new section in the spec file for this purpose:
+
+.. code-block:: spec
+
+    # ========== libdnf5-plugin-template ==========
+
+    %if %{with plugin_template}
+    %package -n libdnf5-plugin-template
+    Summary:        Libdnf5 template plugin
+    License:        LGPL-2.1-or-later
+    Requires:       libdnf5%{?_isa} = %{version}-%{release}
+
+    %description -n libdnf5-plugin-template
+    Include a more descriptive message about your plugin here.
+
+    %files -n libdnf5-plugin-template
+    %{_libdir}/libdnf5/plugins/template.*
+    %config %{_sysconfdir}/dnf/libdnf5-plugins/template.conf
+    %endif
+
+Consider including a documentation man page describing your plugin's functionality.
+Ensure interconnection in these places:
+
+* ``doc/libdnf5_plugins/template.8.rst``: Add a new man page for your plugin with the respective name.
+* ``doc/libdnf5_plugins/index.rst``: Add a reference to the new plugin to the LIBDNF5 plugins page.
+* ``doc/CMakeLists.txt`` and ``doc/conf.py.in``: Integrate with Sphinx documentation.
+* ``dnf5.spec``: Include the new man page in your newly created package.
+
+Debugging Tips
+--------------
+
+To test your freshly built plugin, redirect dnf5 to load your plugin
+configuration by setting the ``LIBDNF_PLUGINS_CONFIG_DIR`` environmental
+variable to your directory with the plugin's configuration, e.g.,
+``LIBDNF_PLUGINS_CONFIG_DIR=/home/user/dnf5/libdnf5-plugins/template.``
+This can also be overridden by the ``pluginconfpath`` configuration option.
+
+Next, configure the directory where the plugin binaries are located using
+the ``pluginpath`` configuration option.
+
+For effective interception of your plugin code at defined breakpoints, ensure
+you are building the project with debugging symbols (``-DCMAKE_BUILD_TYPE=Debug``)."
```

Comment:
can these tips be a single tip section to deduplicate info in the documentation?